Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1

```diff
@@ -200,12 +208,29 @@ $servicePackageFile = Find-Files -SearchPattern "$CsPkg"
 Write-Host "servicePackageFile= $servicePackageFile"
 $servicePackageFile = Get-SingleFile $servicePackageFile $CsPkg
 
-Write-Host "Get-AzureService -ServiceName $ServiceName -ErrorAction SilentlyContinue"
-$azureService = Get-AzureService -ServiceName $ServiceName -ErrorAction SilentlyContinue
+Write-Host "Get-AzureService -ServiceName $ServiceName -ErrorAction SilentlyContinue  -ErrorVariable azureServiceError"
+$azureService = Get-AzureService -ServiceName $ServiceName -ErrorAction SilentlyContinue  -ErrorVariable azureServiceError
+
+if($azureServiceError){
+   $azureServiceError | ForEach-Object { Write-Warning $_.Exception.ToString() }
+}   
+
+   
 if (!$azureService)
-{
-    Write-Host "New-AzureService -ServiceName $ServiceName -Location $ServiceLocation"
-    $azureService = New-AzureService -ServiceName $ServiceName -Location $ServiceLocation
+{    
+    $azureService = "New-AzureService -ServiceName `"$ServiceName`""
+    if($NewServiceAffinityGroup) {
+        $azureService += " -AffinityGroup `"$NewServiceAffinityGroup`""
+    }
+    elseif($ServiceLocation) {
+         $azureService += " -Location `"$ServiceLocation`""
```

Comment:
usually ServiceLocation shouldn't have any special characters, , I think it's fine if it blows away